Hints.py

```diff
@@ -708,6 +708,10 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
     # rebuild hint exclusion list
     hintExclusions(world, clear_cache=True)
 
+    #rebuild named item pool
+    world.settings.item_hints += world.item_added_hint_types["named-item"]
```

Comment:
This PR only fixes the duplication when generating a single seed. If you try to generate more than one, `world.settings.item_hints` will continue to be added on to with the named-item hints until it errors out.